source/includes/library-reference/_config.html.md

```diff
@@ -0,0 +1,109 @@
+# Config Class
+
+Used to manage bonsai configuration environments.
+Config files can be either specified in the users home directory
+or in a local directory. In addition, configuration parameters can be parsed from
+the command line.
+
+Example `~/.bonsai` config file:
+    
+```ini
+[DEFAULT]
+username = admin
+accesskey = None
+profile = dev
+
+[dev]
+url = http://localhost:5000
+username = admin
+accesskey = 00000000-1111-2222-3333-000000000001
+
+[alpha]
+url = https://alpha-api.int.bons.ai
+username = mikest
+```
+
+The `profile` key can be used to switch between different profiles in
+the same configuration file.
+
+## Config(profile)
+
+Constructs a default configuration.
+- `profile` Name of the default profile. Default value is an empty string.
+
+Configurations are stored in `~/.bonsai` and `./.bonsai` configuration files.
+The local configuration file will override settings in the users home directory configuration file.
+    
+```python
+config = Config(sys.argv)
+print(config)
+```
+
+```cpp
+int main(int argc, char** argv) {
+    auto config = Config(argc, argv);
+    std::cout << config << std::endl;
+}
+```
+
+## Config(argc, argv, profile)
+
+Constructs a config by looking in the configuration files and parsing the command line arguments.
+In Python, `argc` is not necessary.
+
+Arguments:
+
+- `argc` ...as passed to `int main(int argc, char** argv)`.
+- `argv` ...same.
+- `profile` Name of the default profile.
+
+Example command line arguments:
+
+- `--accesskey=00000000-1111-2222-3333-000000000001`
+- `--username=admin`
+- `--url=http://localhost:32802`
+
+Unrecognized arguments will be ignored.
+
+## accesskey()
+
+Server authentication token.
+Obtained from the bonsai server. you will need to set it in your config.
+
+## username()
+
+Account user name.
+The account you signed up with.
+
+## url()
+
+Server URL.
+Address and port number of the bonsai server. Normally you should not need to change this.
+
+## proxy()
+
+Proxy Server.
+Address and port number of the proxy server to connect through.
+
+## brain()
+
+BRAIN name.
+Name of the BRAIN on the server.
+
+## predict()
+
+Simulator mode.
+The mode simulators will run in, true if running for prediction, false for training.
+
+## brain_version()
```

Comment:
why `brain_version` vs `version`?

Reply:
because the class is `Config` and it's not the `Config.version()`